src/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useContext, useEffect, useReducer } from "react"
+import { useDidMount } from "Utils/Hooks/useDidMount"
+import { hasFilters } from "./Utils/hasFilters"
+import { isDefaultFilter } from "./Utils/isDefaultFilter"
+import { rangeToTuple } from "./Utils/rangeToTuple"
+import { urlFragmentFromState } from "./Utils/urlFragmentFromState"
+
+export interface Filters {
+  acquireable?: boolean
+  at_auction?: boolean
+  color?: string
+  for_sale?: boolean
+  height: string
+  inquireable_only?: boolean
+  keyword?: string
+  major_periods: string[]
+  medium?: string
+  offerable?: boolean
+  page: number
+  partner_id?: string
+  price_range: string
+  sort: string
+  width: string
+}
+
+export const initialFilterState = {
+  height: "*-*",
+  major_periods: [],
+  page: 1,
+  price_range: "*-*",
+  sort: "-decayed_merch",
+  width: "*-*",
+}
+
+export const FilterContext = React.createContext<{
+  filters: Filters
+  hasFilters: boolean
+  isDefaultValue: (name: string) => boolean
+  rangeToTuple: (name: string) => [number, number]
+  resetFilters: () => void
+  setFilter: (name: string, value: any) => void
+  unsetFilter: (name: string) => void
+}>({
+  filters: initialFilterState,
+  hasFilters: false,
+  isDefaultValue: null,
+  rangeToTuple: null,
+  resetFilters: null,
+  setFilter: null,
+  unsetFilter: null,
+})
+
+export const FilterContextProvider: React.FC<
+  Filters & { children: React.ReactNode }
+> = ({
+  children,
+  height = "*-*",
+  major_periods = [],
+  medium = "",
+  page = 1,
+  price_range = "*-*",
+  sort = "-decayed_merch",
+  width = "*-*",
+  ...rest
+}) => {
+  const [state, dispatch] = useReducer(filterReducer, {
+    height,
+    major_periods,
+    medium,
+    page,
+    price_range,
+    sort,
+    width,
+    ...rest,
+  })
+  const isMounted = useDidMount()
+  // const updateOnChange = Object.keys(state).map(prop => state[prop])
+
+  useEffect(() => {
+    if (isMounted) {
+      const queryString = urlFragmentFromState(state)
+      window.history.pushState(
+        {},
+        null,
+        `${window.location.pathname}?${queryString}`
```

Comment:
Long-term this URL construction algorithm is another thing that will vary across apps. Most can use this string format, but the [Collect app's format](https://github.com/artsy/reaction/blob/9555125ef04e9af1f23a45990ec41cd893b94341/src/Apps/Collect/urlBuilder.ts#L15) is slightly different - because a selected medium factors in as part of the path, not the querystring.

Reply:
I think the way to go here is to tap into `farce` directly, and dispatch update events from there. This is a bit lower-level than the router (the router plugs into it) and it's more global. (I know we've gone down the "use the router" path before but I still have a hunch that we can make it work. If the component exists outside of a formal app (meaning, there's no router context available within the tree) then we can defer to the `window`. (Just brainstorming; this is a `TODO` for later.)